### DISCUSSION TITLE
What is the difference between state and notifier?

### DISCUSSION THREAD

**Discussion by anmol-me** [rrousselGit/riverpod#1824] [Q&A]
```
final stateProvider = StateProvider<bool>((ref) => false);
```

```
StateController<bool> stateController = ref.watch(stateProvider.state);

StateController<bool> notifierController = ref.watch(stateProvider.notifier);
```

Both stateProvider.state and stateProvider.notifier seems to be the same. Why prefer one over the other?

**Comment by rrousselGit** [comment]
`.state` exists mostly for an easy migration from 0.x.y versions to 1.x.y versions, where the behavior of `ref.watch(stateProvider)` changed.

For new usages, use `.notifier` directly.
